airflow/utils/email.py

```diff
@@ -85,7 +85,18 @@ def send_email_smtp(
 
     >>> send_email('test@example.com', 'foo', '<b>Foo</b> bar', ['/dev/null'], dryrun=True)
     """
-    smtp_mail_from = conf.get('smtp', 'SMTP_MAIL_FROM')
+    from_email = kwargs.get('from_email') or os.environ.get('SMTP_MAIL_FROM')
+    if not from_email and conf.has_option('smtp', 'smtp_mail_from'):
+        from_email = conf.get('smtp', 'smtp_mail_from')
+
+    from_name = kwargs.get('from_name') or os.environ.get('SMTP_MAIL_SENDER')
+    if not from_name and conf.has_option('smtp', 'smtp_mail_sender'):
+        from_name = conf.get('smtp', 'smtp_mail_sender')
+
+    if from_email:
+        smtp_mail_from = formataddr((from_name, from_email))
+    else:
+        smtp_mail_from = None
```

Comment:
This repetition seems wrong. Maybe we should only read conf here and keep callers passin in `None`.

Reply:
Are you referring to `from_email` and `smtp_mail_from`? If `None` is passed to `formataddr` as email then the error message is very unclear

Reply:
Sorry I was unclear. I was referring to this almost exact logic being repeated here and in both the aws and sendgrid providers, and it may be better to refactor them to put the logic into `build_mime_message` or a separate function called in those places instead (by parametrizing the config keys). Something like